Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM node:20-alpine AS base
+
+# Stage 1: Install dependencies
+FROM base AS installer
+
+WORKDIR /usr/src/app
+RUN apk add --no-cache libc6-compat
+COPY package.json package-lock.json ./
+COPY prisma ./prisma
+
+RUN \
+    if [ -f package-lock.json ]; then \
+    npm ci --ignore-scripts; \
+    else \
+    echo "package-lock.json not found" && exit 1; \
+    fi
+
+# Stage 2: Build stage
+FROM base AS builder
+WORKDIR /usr/src/app
+
+COPY --from=installer /usr/src/app/node_modules ./node_modules
+COPY --from=installer /usr/src/app/prisma ./prisma
+
+RUN DATABASE_URL=$DATABASE_URL npx prisma generate
```

Comment:
providing database url is not necessary for prisma generate. You can add `npx prisma generate` in the same line when you run `npm run build`